<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
-->
<launch>
  <include file="$(find tb_waiter)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find tb_waiter)/launch/includes/safety_controller.launch.xml"/>

  <arg name="odom_topic" default="odom" />

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find tb_waiter)/param/move_base/move_base/move_base_params.yaml" command="load" />

    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
</launch>
